plugin/pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+	schedulertesting "k8s.io/kubernetes/plugin/pkg/scheduler/testing"
+)
+
+// sortableAntiAffinityTerms lets us to sort anti-affinity terms.
+type sortableAntiAffinityTerms struct {
+	Items []matchingPodAntiAffinityTerm
+}
+
+// Less establishes some ordering between two matchingPodAntiAffinityTerms for
+// sorting.
+func (s *sortableAntiAffinityTerms) Less(i, j int) bool {
+	t1, t2 := s.Items[i], s.Items[j]
+	return t1.node.Name < t2.node.Name ||
+			(t1.node.Name == t2.node.Name && len(t1.term.Namespaces) < len(t2.term.Namespaces)) ||
+			(t1.node.Name == t2.node.Name && len(t1.term.Namespaces) == len(t2.term.Namespaces) && t1.term.TopologyKey < t2.term.TopologyKey) ||
+			(t1.node.Name == t2.node.Name && len(t1.term.Namespaces) == len(t2.term.Namespaces) &&
+					t1.term.TopologyKey == t2.term.TopologyKey &&	len(t1.term.LabelSelector.MatchLabels) < len(t2.term.LabelSelector.MatchLabels))
+}
+func (s *sortableAntiAffinityTerms) Len() int { return len(s.Items) }
+func (s *sortableAntiAffinityTerms) Swap(i, j int) {
+	s.Items[i], s.Items[j] = s.Items[j], s.Items[i]
+}
+
+var _ = sort.Interface(&sortableAntiAffinityTerms{})
+
+func sortAntiAffinityTerms(terms map[string][]matchingPodAntiAffinityTerm) {
+	for k, v := range terms {
+		sortableTerms := sortableAntiAffinityTerms{v}
+		sort.Sort(&sortableTerms)
+		terms[k] = sortableTerms.Items
+	}
+}
+
+// sortablePods lets us to sort pods.
+type sortablePods struct {
+	Items []*v1.Pod
+}
+
+func (s *sortablePods) Less(i, j int) bool {
+	if s.Items[i].Name < s.Items[j].Name {
```

Comment:
This isn't necessarily good sorting function.
You can have two pods with the same name but in different namespaces. And those aren't equal.

Should be:
```
return s[i].Namespace < s[j].Namespace || (s[i].Namespace == s[j].Namespace && s[i].Name < s[j].Name)
```